content/file/utils.go

```diff
@@ -214,7 +214,12 @@ func extractTarDirectory(dirPath, dirName string, r io.Reader, buf []byte) error
 		}
 
 		// Change access time and modification time if possible (error ignored)
-		os.Chtimes(filePath, header.AccessTime, header.ModTime)
+		_ = os.Chtimes(filePath, header.AccessTime, header.ModTime)
+
+		// Restore full mode bits (error ignored)
+		if preserveModeBits && (header.Typeflag == tar.TypeReg || header.Typeflag == tar.TypeDir) {
+			_ = os.Chmod(filePath, os.FileMode(header.Mode))
```

Comment:
Since the user set the option, they may want to get an error if the permission cannot be kept. I think we can return a dedicated error type for users to catch.
What do you think?

Reply:
Yea, I like that.
Fixed